src/App.tsx

```diff
@@ -1,61 +1,149 @@
 import './App.scss';
+import { TodoList } from './components/TodoList';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+import { useState } from 'react';
+
+export type User = {
+  id: number;
+  name: string;
+  username: string;
+  email: string;
+};
+
+export type Todo = {
+  id: number;
+  title: string;
+  completed: boolean;
+  userId: number;
+  user?: User | null;
+};
+
+const getUserById = (userId: number): User | null => {
+  return (
+    usersFromServer.find(userFromServer => userId === userFromServer.id) || null
+  );
+};
+
+export const todos: Todo[] = todosFromServer.map(todo => {
+  return { ...todo, user: getUserById(todo.userId) };
+});
 
 export const App = () => {
+  const [visibleTodos, setVisibleTodos] = useState(todos);
+  const [title, setTitle] = useState('');
+  const [selectedUserId, setSelectedUserId] = useState('0');
+  const [titleError, setTitleError] = useState(false);
+  const [userError, setUserError] = useState(false);
+
+  const handleSubmit = (event: React.FormEvent) => {
+    event.preventDefault();
+
+    let isValid = true;
+
+    if (title === '') {
+      setTitleError(true);
+      isValid = false;
+    } else {
+      setTitleError(false);
+    }
+
+    if (selectedUserId === '0') {
+      setUserError(true);
+      isValid = false;
+    } else {
+      setUserError(false);
+    }
+
+    if (!isValid) {
+      return;
+    }
+
+    const getId = () => {
+      if (visibleTodos.length === 0) {
+        return 1;
+      }
+
+      return Math.max(...visibleTodos.map(todo => todo.id)) + 1;
```

Comment:
The `getId` function calculates the new ID by finding the maximum ID in `visibleTodos` and adding 1. This approach can be inefficient for large arrays. Consider using a more efficient method if performance becomes an issue.